src/drivers/block_dev/block_dev.c

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file
+ * @brief
+ *
+ * @date 31.07.2012
+ * @author Andrey Gazukin
+ */
+
+#include <errno.h>
+#include <libgen.h>
+
+#include <drivers/block_dev.h>
+
+extern struct idesc_ops idesc_bdev_ops;
+extern int devfs_add_block(struct block_dev *dev);
+extern int devfs_del_block(struct block_dev *dev);
+
+struct block_dev *block_dev_create(const char *path, const struct block_dev_driver *driver, void *privdata) {
+	struct block_dev *bdev;
+	struct dev_module *devmod;
+
+	bdev = block_dev_create_common((char *)basename((char *)path), driver, privdata);
```

Comment:
Can we now merge `block_dev_create()` and `block_dev_create_common()` into one function?

Reply:
Yes, I think so